backend/backend.go

```diff
@@ -122,3 +122,11 @@ func UnmarshalHistoryEvent(bytes []byte) (*HistoryEvent, error) {
 	}
 	return e, nil
 }
+
+func BuildStatusSet(statuses []api.OrchestrationStatus) map[api.OrchestrationStatus]struct{} {
+	statusSet := make(map[api.OrchestrationStatus]struct{})
```

Comment:
Small optimization that allows pre-allocating just the amount of memory we need

```suggestion
	statusSet := make(map[api.OrchestrationStatus]struct{}, len(statuses))
```